app/models/order.rb

```diff
@@ -18,11 +18,16 @@ def set_defaults
     self.state = "Created"
   end
 
-  def finalize_order
+  def transition_state
     item_states = order_items.collect(&:state)
+
     if item_states.include?('Failed')
       self.state = "Failed"
-    elsif item_states.include?('Completed') && item_states.count == order_items.count
+    elsif item_states.include?('Denied')
+      self.state = "Denied"
+    elsif item_states.all? { |state| state == "Approved" }
```

Comment:
I originally thought that there was an easier way to write this, but unfortunately it doesn't exist until Ruby 2.5 *or* we could use this ugly way:
`item_states.all?(&"Approved".method(:==))`

It's cool, but, not really readable, haha.

Once we're confirmed on 2.5 we could potentially just change it to:
`item_states.all?(/Approved/)`

For now, I just figured this was an interesting thing to share 😛